src/Honeycomb.OpenTelemetry/ConsoleLinksExporter.cs

```diff
@@ -0,0 +1,103 @@
+using OpenTelemetry;
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Honeycomb.OpenTelemetry
+{
+    /// <summary>
+    /// Writes links to the Honeycomb for all root spans
+    /// </summary>
+    public class ConsoleLinkExporter : BaseExporter<Activity>
+    {
+        private readonly HoneycombOptions _options;
+        private string _teamSlug;
+        private string _environmentSlug;
+
+        private bool IsEnabled = false;
+
+        /// <summary>
+        /// Initializes the <see cref="ConsoleLinkExporter" /> class
+        /// </summary>
+        /// <param name="options">Settings for Link generation</param>
+        public ConsoleLinkExporter(HoneycombOptions options)
+        {
+            _options = options;
+            InitTraceLinkParameters();
+        }
+
+        private void InitTraceLinkParameters()
+        {
+            if (string.IsNullOrEmpty(_options.ApiKey))
+                return;
+
+            var httpClient = new HttpClient();
+            httpClient.BaseAddress = new Uri("https://api.honeycomb.io/1/auth");
+            httpClient.DefaultRequestHeaders.Add("X-Honeycomb-Team", _options.ApiKey);
+
+            var response = httpClient.GetAsync("").GetAwaiter().GetResult();
+            if (!response.IsSuccessStatusCode) {
+                Console.WriteLine("Didn't get a valid response from Honeycomb");
+                return;
+            }
+
+            var responseString = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+            var authResponse = JsonSerializer.Deserialize<AuthResponse>(responseString);
+            _environmentSlug = authResponse.Environment.Slug;
+            _teamSlug = authResponse.Team.Slug;
+            if (string.IsNullOrEmpty(_environmentSlug) ||
+                string.IsNullOrEmpty(_teamSlug)) {
+                Console.WriteLine("Team or Environment wasn't returned");
+                return;                
+            }
+            IsEnabled = true;
+        }
+
+        /// <inheritdoc />
+        public override ExportResult Export(in Batch<Activity> batch)
+        {
+            if (!IsEnabled)
+                return ExportResult.Success;
+
+            foreach (var activity in batch)
+            {
+                if (string.IsNullOrEmpty(activity.ParentId))
+                {
+                    Console.WriteLine($"Trace Emitted for {activity.DisplayName}");
+                    Console.WriteLine($"TraceLink: {GetTraceLink(activity.TraceId.ToString())}");
+                }
+            }
+            return ExportResult.Success;
+        }
+
+        private string GetTraceLink(string traceId)
+        {
+            var dataset = _options.ServiceName;   
+            return $"http://ui.honeycomb.io/{_teamSlug}/environments/{_environmentSlug}/datasets/{dataset}/trace?trace_id={traceId}";

```

Comment:
Same as above, we should consider allowing trace links to be created for classic (non-env) based users too.

```suggestion
            return $"{_options.TracesEndpoint}/{_teamSlug}/environments/{_environmentSlug}/datasets/{dataset}/trace?trace_id={traceId}";

```

Reply:
If I have a way of knowing that it's a classic just from the ApiKey or settings, then I'm happy to add it.